network/gossip.go

```diff
@@ -43,6 +38,8 @@ func (t *Topic) Publish(obj proto.Message) error {
 }
 
 func (t *Topic) Subscribe(handler func(obj interface{})) error {
+	bufferSize := int(t.bufferSize)
```

Comment:
Topic needs only int bufferSize. Why not convert in NewTopic?

Reply:
This was only draft PR, nothing ready for review, will ping you once i prepare PR, thank you anyways for the review!